### ISSUE TITLE
there is no check if diff >= 0

### ISSUE THREAD

**Issue by yakovyarmo** [postcss/autoprefixer#927]
https://github.com/postcss/autoprefixer/blob/aafd9ee60899de869b10ed7d750e2986be6ddec2/lib/declaration.js#L91

sometimes diff is negative for me and then i get the following:
RangeError: Invalid array length

**Comment by ai** [comment]
I need input CSS to reproduce the error.

**Comment by yakovyarmo** [comment]
the css is

```
.mck-sidebox-launcher .mck-button-launcher {
        box-shadow: 0 0 4px rgba(0, 0, 0, .14), 0 4px 8px rgba(0, 0, 0, .28);
        background-size: 22px 22px;
        box-sizing: content-box;
        background-repeat: no-repeat;
        border-radius: 50%;
        border-style: solid;
        border-width: 1px;
        position: absolute;
        bottom: 0;
        right: 0;
        width: 48px;
        height: 48px;
        cursor: pointer;
        outline: none;
        background-position: center;
        background-color: #0071b2;
        border-color: #00498a;
        -webkit-user-drag: none;
        -webkit-transform: scale(1) rotate(360deg);
        transform: scale(1) rotate(360deg);
        -webkit-transition: -webkit-transform 150ms cubic-bezier(.4, 0, 1, 1);
        transition: transform 150ms cubic-bezier(.4, 0, 1, 1);
    }
```
and the conifg:

Autoprefixer({'browsers': ['ie >=9']})

**Comment by Semigradsky** [comment]
Is not reproduced for me

**Comment by ai** [comment]
@yakovyarmo what Autoprefixer version do you use and what is full error stacktrace?

**Comment by yakovyarmo** [comment]
How do I do that?

**Comment by ai** [comment]
1. `npm ls | grep autoprefixer`.
2. Reproduce error, copy full error message from console.

**Comment by yakovyarmo** [comment]
yarn list --pattern autoprefixer
yarn list v1.2.1
├─ autoprefixer@7.1.5
└─ cssnano@3.10.0
   └─ autoprefixer@6.7.7

RangeError: Invalid array length
    at TransformDecl.calcBefore (C:\Users\yakoyarmo\PhpstormProjects\limudnaim-rackspace\vue\node_modules\autoprefixer\lib\declaration.js:146:19)
    at TransformDecl.insert (C:\Users\yakoyarmo\PhpstormProjects\limudnaim-rackspace\vue\node_modules\autoprefixer\lib\declaration.js:189:39)
    at TransformDecl.insert (C:\Users\yakoyarmo\PhpstormProjects\limudnaim-rackspace\vue\node_modules\autoprefixer\lib\hacks\transform-decl.js:96:50)
    at TransformDecl.add (C:\Users\yakoyarmo\PhpstormProjects\limudnaim-rackspace\vue\node_modules\autoprefixer\lib\declaration.js:221:21)
    at TransformDecl.process (C:\Users\yakoyarmo\PhpstormProjects\limudnaim-rackspace\vue\node_modules\autoprefixer\lib\prefixer.js:170:22)
    at TransformDecl.process (C:\Users\yakoyarmo\PhpstormProjects\limudnaim-rackspace\vue\node_modules\autoprefixer\lib\declaration.js:235:52)
    at C:\Users\yakoyarmo\PhpstormProjects\limudnaim-rackspace\vue\node_modules\autoprefixer\lib\processor.js:139:37
    at C:\Users\yakoyarmo\PhpstormProjects\limudnaim-rackspace\vue\node_modules\postcss\lib\container.js:192:28
    at C:\Users\yakoyarmo\PhpstormProjects\limudnaim-rackspace\vue\node_modules\postcss\lib\container.js:148:26
    at Rule.each (C:\Users\yakoyarmo\PhpstormProjects\limudnaim-rackspace\vue\node_modules\postcss\lib\container.js:114:22

[style.css.txt](https://github.com/postcss/autoprefixer/files/1401831/style.css.txt)

**Comment by ai** [comment]
Fixed 8d457bd

**Comment by ai** [comment]
Released in 7.1.6.
